kubernetes/provider.go

```diff
@@ -202,48 +202,52 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
 	if err != nil {
 		return nil, err
 	}
+
+	hasConfig := false
```

Comment:
An alternative to having this variable would have been to check if `cfg` points to an empty struct.

Reply:
any reason why you did not go with that route? It's seems like a much more maintainable approach.

Reply:
It should work, I'm going to add unit tests and experiment with it.